addons/product/models/product_template.py

```diff
@@ -179,70 +179,72 @@ def _compute_currency_id(self):
     def _compute_cost_currency_id(self):
         self.cost_currency_id = self.env.company.currency_id.id
 
+    def _compute_product_variant_field(self, fname, default=False):
+        """Computes a field from the template's product.product
+
+        Equals to product_variant_ids[fname] if it's a single variant product.
+        Otherwise, sets the value specified in ``default``.
+        It's used to compute fields like barcode, weight, volume..
+        """
+        for template in self:
+            variant_count = len(template.product_variant_ids)
+            if variant_count == 1:
+                template[fname] = template.product_variant_ids[fname]
+            elif variant_count == 0 and self.env.context.get("active_test", True):
+                # If the product has no active variants, retry without the active_test
+                template.with_context(active_test=False)._compute_product_variant_field(fname, default=default)
+            else:
+                template[fname] = default
+
+    def _set_product_variant_field(self, fname):
+        """Set a field to the template's product.product
+
+        Only if it's a single variant product.
+        It's used to set fields like barcode, weight, volume..
+        """
+        for template in self:
+            if len(template.product_variant_ids) == 1:
+                template.product_variant_ids[fname] = template[fname]
+
     @api.depends_context('company')
-    @api.depends('product_variant_ids', 'product_variant_ids.standard_price')
+    @api.depends('product_variant_ids.standard_price')
     def _compute_standard_price(self):
-        # Depends on force_company context because standard_price is company_dependent
-        # on the product_product
-        unique_variants = self.filtered(lambda template: len(template.product_variant_ids) == 1)
-        for template in unique_variants:
-            template.standard_price = template.product_variant_ids.standard_price
-        for template in (self - unique_variants):
-            template.standard_price = 0.0
+        # Depends on force_company context because standard_price is company_dependent on the product_product
```

Comment:
```suggestion
        # Depends on force_company context because standard_price is company_dependent
        # on the product_product
```

Keep it on two lines pls, to respect a max line length of 100 characters